kanopy-platform/gateway-certificate-controller
# Istio 1.21 Upgrade

Upgrades istio go.mod dependencies to v1.21.6

There are lots of [dependabot alerts](https://github.com/kanopy-platform/gateway-certificate-controller/security/dependabot), this should fix most of them.

### Testing

1. With kind cluster up, `make kind-istio`
2. `skaffold dev`
3. Apply the following gateway
```
---
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  name: httpbin-gateway
  namespace: devops
  labels:
    "v1beta1.kanopy-platform.github.io/istio-cert-controller-inject-simple-credential-name": "true"
spec:
  selector:
    istio: istio-ingressgateway-public
  servers:
    - port:
        number: 443
        name: https
        protocol: HTTPS
      hosts:
        - "devops/httpbin.example.com"
      tls:
        mode: SIMPLE
```
4. Notice that controller logs show it received request. `kubectl get gateway -n devops httpbin-gateway -oyaml` shows tls.credentialName injected. `k get cert -n routing devops-httpbin-gateway-https -oyaml` whows cert was created and ready.
5. Run `k delete -n devops gateway httpbin-gateway`, after a few seconds see that Cert was cleaned up by [garbage collection controller](https://github.com/kanopy-platform/gateway-certificate-controller/blob/main/docs/controllers/garbage_collection.md) `k get cert -n routing devops-httpbin-gateway-https`
